### DISCUSSION TITLE
Default new entries to use current locale, not site default

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#9387] [Ideas]
> Created by: Chris (chris@paragon.net.uk) on 2016/08/01 12:35:09 +0000
> Votes at time of UserVoice import: 4

-----------------------------------

It would be great if when creating a new entry, it defaulted to the current locale, rather than the site default. For example, if I have 3 locales ('en_gb' (default), 'de', 'fr') and I switch to the 'fr' locale using the dropdown above the entries table - when I click 'New Entry' I would expect/like to author a new entry in that locale, rather than 'en_gb'.

The fix is easy - changing line 10805 in craft.js from:

return 'href="'+Craft.getUrl('entries/'+section.handle+'/new')+'"';

to

return 'href="'+Craft.getUrl('entries/'+section.handle+'/new/'+this.locale)+'"';

**Comment by robmcfadden** [comment]
It would also be nice if when saving an entry, it would stay on the current site, rather than switching back to the default. Craft 2 did this for locales.

@takobell - This additional request feels related. Should I create a new issue or just leave it here in the comment? Thanks!

**Comment by brandonkelly** [comment]
@robmcfadden if that’s the case it is a bug. Can you please post as a new issue?
